### DISCUSSION TITLE
Changing character mods dynamically?

### DISCUSSION THREAD

**Discussion by spelled-ayayron** [easy-games/chickynoid#20] [Q&A]
Hi, I noticed in the example that the player's character mod is set to "NicerHumanoid" when they initially load in. I want to create multiple character mods with different movement parameters that players can switch between during gameplay. I tried testing this by calling `playerRecord:SetCharacterMod()` again after the first time it is called, but the change doesn't occur until I despawn the player.

**Comment by TheGibusGuy** [comment]
I realize this answer is absurdly late, but I think I have an idea. The simulation "constants" aren't really constant. You can change them whenever you want. Perhaps you could make a dict with the different sets of movement parameters, and have them be applied in your MoveType moduleScript.

Something kinda like this:

```
local module = {}
local path = game.ReplicatedFirst.Packages.Chickynoid
local MathUtils = require(path.Simulation.MathUtils)
local Enums = require(path.Enums)

local isServer = false
if (game:GetService("RunService"):IsServer()) then isServer = true end

function module:Setup(simulation)
	self:ModifySimulation(simulation)
end

module.parameters = {}
paramters["Knight"] = { maxSpeed = 32, accel = 4 }
paramters["Archer"] = { maxSpeed = 48, accel = 6 }
paramters["Cavalry"] = { maxSpeed = 64, accel = 8 }

function module:ModifySimulation(simulation)
    	simulation:RegisterMoveState("Walking", self.ActiveThink, nil, nil, nil)
    	simulation:SetMoveState("Walking")
	
	simulation.state.parameterSet = "Knight" -- a serverMod could be allowed to change this
end

function module:ActiveThink(cmd)
	
	self.constants = self.parameters[self.state.parameterSet]

	-- the rest of the movement code
end

return module
```

**Reply by spelled-ayayron** [reply]
This is a neat approach to handling changing the simulation constants. I did have my own method working on my fork but didn't make a PR for it since it was pretty niche and kinda hacky. I'll try implementing this and if it ends up neater will make a PR from my fork sometime!
